### PR TITLE
fix: temporary fix for `api` dependency

### PR DESCRIPTION
**Pull request by sergio-mena**
The current `go.mod` points to CometBFT version `v1.0.0-alpha.1`. After merging `feature/pbts` onto `main` there are changes in protobufs that aren't included in that version.

This PR add a temporary fix that will work until:
* we tag a new `alpha` version of CometBFT
* we tag the final `api` version as part of the release process

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments
- [ ] Title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -166,21 +166,26 @@ backport branch (see above). Otherwise:
      release.
    * Ensure that `UPGRADING.md` is up-to-date and includes notes on any breaking
      changes or other upgrading flows.
-4. Prepare the versioning:
+4. Check the dependency to `github.com/cometbft/cometbft/api` in the `go.mod`
+   directory. If it does not point to an official api version, edit it
```
Comment (sergio-mena) [inline RELEASES.md:170]:
```suggestion
   file. If it does not point to an official api version, edit it
```

File: RELEASES.md [hunk 2]
```diff
@@ -194,6 +199,11 @@ Before performing these steps, be sure the
 1. Start on the backport branch (e.g. `v2.x`)
 2. Run integration tests (`make test_integrations`) and the E2E nightlies.
 3. Prepare the release:
+   * Check the dependency to `github.com/cometbft/cometbft/api` in the `go.mod`
+     directory. If it does not point to an official api version, edit it
```
Comment (sergio-mena) [inline RELEASES.md:203]:
```suggestion
     file. If it does not point to an official api version, edit it
```

File: RELEASES.md [hunk 3]
```diff
@@ -226,6 +236,11 @@ To create a patch release:
 1. Checkout the long-lived backport branch: `git checkout v2.x`
 2. Run integration tests (`make test_integrations`) and the nightlies.
 3. Check out a new branch and prepare the release:
+   * Check the dependency to `github.com/cometbft/cometbft/api` in the `go.mod`
+     directory. If it does not point to an official api version, edit it
```
Comment (sergio-mena) [inline RELEASES.md:240]:
```suggestion
     file. If it does not point to an official api version, edit it
```

File: go.mod [hunk 4]
```diff
@@ -41,7 +41,7 @@ require (
 	github.com/btcsuite/btcd/btcutil v1.1.5
 	github.com/cometbft/cometbft-db v0.11.0
 	github.com/cometbft/cometbft-load-test v0.1.0
-	github.com/cometbft/cometbft/api v1.0.0-alpha.1
+	github.com/cometbft/cometbft/api v0.0.0-20240311183503-2f0dc67ee8d5
```
Comment (lasarojc) [inline go.mod:44]: why v0?
I think it should signal that it is v1 related. can we remove this tag later, from the registry?

**Review by lasarojc — APPROVED** [review]

**Review by melekes — APPROVED** [review]

File: go.mod (hunk 4 above)
Comment (sergio-mena) [reply go.mod:44]: It's the format added by go itself, when using `go get ...`

**Review by adizere — APPROVED** [review]

**Pull request merged**